python/foglamp/tasks/north/sending_process.py

```diff
@@ -374,884 +284,654 @@ def __init__(self, loop=None):
+                # Handles the sleep time, it is doubled every time up to a limit
+                if slept:
+                    sleep_num_increments += 1
+                    sleep_time *= 2
+                    if sleep_num_increments > self.TASK_SLEEP_MAX_INCREMENTS:
+                        sleep_time = self.TASK_SEND_SLEEP
+                        sleep_num_increments = 1
 
-        except Exception as _ex:
-            _message = _MESSAGES_LIST["e000009"].format(str(_ex))
-            SendingProcess._logger.error(_message)
+            # Checks if the information on the Storage layer needs to be updates
+            if db_update:
+                await self._update_position_reached(update_last_object_id, tot_num_sent)
+        except Exception as ex:
+            SendingProcess._logger.error(_MESSAGES_LIST["e000021"].format(ex))
+            if db_update:
+                await self._update_position_reached(update_last_object_id, tot_num_sent)
+            await self._audit.failure(self._AUDIT_CODE, {"error - on _task_send_data": _message})
             raise
-        return converted_data
 
-    @staticmethod
-    def _transform_in_memory_data_readings(raw_data):
-        """ Transforms readings data retrieved form the DB layer to the proper format
-        Args:
-            raw_data: list of dicts to convert having the structure
-                id         : int  - Row id on the storage layer
-                asset_code : str  - Asset code
-                read_key   : str  - Id of the row
-                reading    : dict - Payload
-                user_ts    : str  - Timestamp as str
-        Returns:
-            converted_data: converted data
-        Raises:
+    async def _load_data_into_memory_audit(self, last_object_id):
+        """ Extracts from the DB Layer data related to the statistics audit into the memory
+        Todo: TO BE IMPLEMENTED
         """
-        converted_data = []
+        raw_data = None
+        try:
+            # Temporary code
+            if self._module_template != "":
+                raw_data = ""
+            else:
+                raw_data = ""
+        except Exception:
+            SendingProcess._logger.error(_MESSAGES_LIST["e000000"])
+            raise
+        return raw_data
 
+    @staticmethod
+    def _transform_in_memory_data_statistics(raw_data):
+        converted_data = []
         try:
             for row in raw_data:
-
-                # Converts values to the proper types, for example "180.2" to float 180.2
-                payload = row['reading']
-                for key in list(payload.keys()):
-                    value = payload[key]
-                    payload[key] = plugin_common.convert_to_type(value)
-
-                # Adds timezone UTC
-                timestamp = apply_date_format(row['user_ts'])
-
+                timestamp = apply_date_format(row['ts'])  # Adds timezone UTC
+                asset_code = row['key'].strip()
```

Comment:
these 2 lines can simply be in new_row construction block, the way `reading` key got the value!